File: 2017/day04_1.py
# pylint: disable=missing-module-docstring, missing-function-docstring


from typing import Any


def solution(content: list[str]) -> Any:
    return len([line for line in content if is_valid(line)])


def is_valid(line: str):
    words = line.strip().split(" ")
    return len(words) == len(set(words))


def main() -> None:
    with open("day04.dat", "r", encoding="utf8") as file:
        print(solution(file.readlines()))


if __name__ == "__main__":
    main()

File: 2017/day04-1.ts
function solution(content: string) {
  return content.split("\n").filter(isValid).length;
}

function isValid(line: string) {
  const words = line.trim().split(" ");
  return words.length === new Set(words).size;
}

Deno.readTextFile("day04.dat")
  .then((content) => content.trim())
  .then(solution)
  .then(console.log);
